packages/typography/src/InlineCode/InlineCode.tsx

```diff
@@ -60,14 +66,14 @@ const InlineCode = React.forwardRef<HTMLElement, InlineCodeProps>(
       </code>
     );
 
-    if (isAnchor) {
+    if (needsWrapper) {
```

Comment:
Don't love the conditional rendering here. I think we're trying to move away from `renderedXYZ` type variables

Reply:
I would recommend something like:
```tsx 
  const { Component } = useInferredPolymorphic(as, rest, Fragment);
  const isFragment = isUndefined(as) // or isEqual(Component, Fragment)
  ...
  <Component {...(isFragment && {
      className: cx(...),
      ...rest
    }
  )}>
    <code>
      ...
    </code>
  </Component>

```

Reply:
Fragments don't accept props though

Reply:
Oops I had the check backwards. You'd need to check `!isFragment && ...`. It's a bit of an ugly solution tbf